core/src/View/Kokkos_BasicView.hpp

```diff
@@ -513,6 +513,74 @@ class BasicView {
             data_handle_type{Impl::get_property<Impl::PointerTag>(arg_prop)},
             arg_mapping) {}
 
+#ifdef KOKKOS_ENABLE_DEBUG_BOUNDS_CHECK
+ private:
+  template <class Arg>
+  KOKKOS_INLINE_FUNCTION static bool is_in_bounds(size_t extent,
+                                                  const Arg arg) {
+    return static_cast<std::size_t>(arg) < extent;
+  }
+
+  template <class T>
+  KOKKOS_INLINE_FUNCTION static bool is_in_bounds(size_t extent,
+                                                  const std::pair<T, T> arg) {
+    return static_cast<std::size_t>(arg.second) <= extent && arg.first >= 0;
```

Comment:
There should also be a check for `arg.first <= arg.second`
```suggestion
    return static_cast<std::size_t>(arg.second) <= extent && arg.first >= 0 &&
           arg.first <= arg.second;
```